xsnippet_api/services/snippet.py

```diff
@@ -0,0 +1,63 @@
+"""
+    xsnippet_api.services.snippet
+    -----------------------------
+
+    Snippet service implements domain business logic for managing
+    snippets. Its purpose is to avoid writing business logic in
+    RESTful API.
+
+    :copyright: (c) 2016 The XSnippet Team, see AUTHORS for details
+    :license: MIT, see LICENSE for details
+"""
+
+import datetime
+
+from aiohttp import web
+import pymongo
+
+
+class Snippet:
+
+    def __init__(self, db):
+        self.db = db
+
+    async def create(self, snippet):
+        snippet = self._normalize(snippet)
+        await self.db.snippets.insert(snippet)
+        snippet['id'] = snippet.pop('_id')
+        return snippet
+
+    async def get(self, *, limit=100):
+        snippets = await self.db.snippets.find()\
+            .sort('_id', pymongo.DESCENDING)\
+            .limit(limit)\
+            .to_list(None)
+
+        for snippet in snippets:
+            snippet['id'] = snippet.pop('_id')
+
+        return snippets
+
+    async def get_one(self, id):
+        snippet = await self.db.snippets.find_one({'_id': id})
+
+        if snippet is None:
+            # TODO: raise business domain exception instead
+            raise web.HTTPNotFound()
+
+        snippet['id'] = snippet.pop('_id')
+        return snippet
+
+    def _normalize(self, snippet):
+        rv = dict({
+            'title': None,
+            'author_id': None,
+            'syntax': 'text',
+            'is_public': True,
+            'tags': [],
+        }, **snippet)
+
+        rv['created_at'] = datetime.datetime.utcnow().replace(microsecond=0)
+        rv['updated_at'] = rv['created_at']
```

Comment:
I think there are at least two strategies of initializing ``updated_at`` field:

1) to set the ``updated_at`` value to be equal to ``created_at`` value on creation of a new entity

2) set it to ``None``

In OpenStack we use the latter. I don't really mind, though: it's still possible to find which snippets have never been updated by looking for snippets with ``updated_at`` == ``created_at``, although there might be an edge case, when a snippet is updated *really* fast after it was created.